tutorials/large/L2_large_link_prediction.py

```diff
@@ -23,62 +23,53 @@
 # Link Prediction Overview
 # ------------------------
 #
-# Link prediction requires the model to predict the probability of
-# existence of an edge. This tutorial does so by computing a dot product
-# between the representations of both incident nodes.
-#
-# .. math::
-#
-#
-#    \hat{y}_{u\sim v} = \sigma(h_u^T h_v)
-#
-# It then minimizes the following binary cross entropy loss.
-#
-# .. math::
+# Unlike node classification predicts labels for nodes based on their
+# local neighborhoods, link prediction assesses the likelihood of an edge
+# existing between two nodes, necessitating different sampling strategies
+# that account for pairs of nodes and their joint neighborhoods.
 #
-#
-#    \mathcal{L} = -\sum_{u\sim v\in \mathcal{D}}\left( y_{u\sim v}\log(\hat{y}_{u\sim v}) + (1-y_{u\sim v})\log(1-\hat{y}_{u\sim v})) \right)
-#
-# This is identical to the link prediction formulation in :doc:`the previous
-# tutorial on link prediction <../blitz/4_link_predict>`.
+# Link prediction requires the model to predict the probability of
+# existence of an edge. This tutorial does so by computing a Mean
+# Reciprocal Rank (MRR) score for each edge, which is a common metric for
+# link prediction.
 #
 
 
 ######################################################################
 # Loading Dataset
 # ---------------
 #
-# This tutorial loads the dataset from the ``ogb`` package as in the
-# :doc:`previous tutorial <L1_large_node_classification>`.
+# `ogbl-citation2` is already prepared as ``BuiltinDataset`` in GraphBolt.
 #
 
 import os
 
 os.environ["DGLBACKEND"] = "pytorch"
-import dgl
+import dgl.graphbolt as gb
 import numpy as np
 import torch
-from ogb.nodeproppred import DglNodePropPredDataset
+import tqdm
 
-dataset = DglNodePropPredDataset("ogbn-arxiv")
+dataset = gb.BuiltinDataset("ogbl-citation2").load()
```

Comment:
nit: Will this be very slow if we use ogbl-citation2?

Reply:
Yes, it takes about 5 min. It's a bit unacceptable

Reply:
replaced with `cora` now